ada_feeding/ada_feeding/behaviors/moveit2/modify_collision_object.py

```diff
@@ -0,0 +1,230 @@
+            ADD or MOVE and the collision object position/orientation do not have a frame_id.
+        position_offset: The offset to apply to the collision object position. Only used if
+            the operation is ADD or MOVE.
+        """
+        # pylint: disable=unused-argument, duplicate-code, too-many-arguments
+        # Arguments are handled generically in base class.
+        super().blackboard_inputs(
+            **{key: value for key, value in locals().items() if key != "self"}
+        )
+
+    @override
+    def setup(self, **kwargs):
+        # Docstring copied from @override
+
+        # pylint: disable=attribute-defined-outside-init
+        # It is okay for attributes in behaviors to be
+        # defined in the setup / initialise functions.
+
+        self.logger.info(f"{self.name} [ModifyCollisionObject::setup()]")
+
+        # Get Node from Kwargs
+        self.node = kwargs["node"]
+
+        # Get the MoveIt2 object.
+        self.moveit2, self.moveit2_lock = get_moveit2_object(
+            self.blackboard,
+            self.node,
+        )
+
+    @override
+    def initialise(self):
+        # Docstring copied from @override
+
+        self.logger.info(f"{self.name} [ModifyCollisionObject::initialise()]")
+
+        # Check that the right parameters have been passed
+        operation = self.blackboard_get("operation")
+        if operation in set(
+            (ModifyCollisionObjectOperation.ADD, ModifyCollisionObjectOperation.MOVE)
+        ):
+            collision_object_position = self.blackboard_get("collision_object_position")
+            collision_object_orientation = self.blackboard_get(
+                "collision_object_orientation"
+            )
+            if collision_object_position is None:
+                raise ValueError(
+                    "The collision object position input key must be specified for ADD "
+                    "and MOVE operations."
+                )
+            if collision_object_orientation is None:
+                raise ValueError(
+                    "The collision object orientation input key must be specified for "
+                    "ADD and MOVE operations."
+                )
+        if operation == ModifyCollisionObjectOperation.ADD:
+            mesh_filepath = self.blackboard_get("mesh_filepath")
+            prim_type = self.blackboard_get("prim_type")
+            dims = self.blackboard_get("dims")
+            if mesh_filepath is None and (prim_type is None or dims is None):
+                raise ValueError(
+                    "If `mesh_filepath` is None, then both `prim_type` and `dims` "
+                    "must be specified."
+                )
+
+    @override
+    def update(self) -> py_trees.common.Status:
+        # Docstring copied from @override
+
+        self.logger.info(f"{self.name} [ModifyCollisionObject::update()]")
+
+        # Get the blackboard inputs for all operations
+        operation = self.blackboard_get("operation")
```

Comment:
Note it may still be useful to add a set of `blackboard_exists` calls even if the input has a default value, as a programmer could make an error and send in an empty BlackboardKey.

If this style of checking is used a lot, it may be useful to add a utility function to BlackboardBehavior.

Reply:
I don't understand the feedback here. If a programmer passes in a wrong value or an empty key, we **_want_** an error to be raised here, right? Isn't that already achieved through `blackboard_get`?

Reply:
Oh NVM, I think what you mean is to fail the behavior if it doesn't exist on the blackboard? I can do that.